### PR TITLE
Dont close the connection when the `IdleState.READER_IDLE` occur when processing a batch.

### PR DESCRIPTION
**Pull request by ph**
When logstash takes more time to process a batch than the `client_inactivity_timeout`,
the server was closing the connection in the middle of working on a batch. This commit changes the behavior,
if logstash is processing a batch and the event occur we wont drop the connection otherwise we will close it.

Fixes: #141, #143

### CODE REVIEW

**Comment by ph** [comment]
@talevy would you mind taking a look at this, since you already know this part of the code?

**Comment by ph** [comment]
The failling travis seems to be a timing issue, with how the integration test make the assertions with the external process.

**Comment by talevy** [comment]
@ph I have a question about the keepAlive. what happens if there are no messages to be processed and we close the connection? will filebeat properly handle the networking in this scenario?

File: src/main/java/org/logstash/beats/BeatsHandler.java [hunk 1]
```diff
@@ -84,8 +84,10 @@ private void writeAck(ChannelHandlerContext ctx, byte protocol, int sequence) {
     }
 
     private void clientTimeout() {
-        logger.debug("Client Timeout");
-        this.context.close();
+        if(!processing.get()) {
```
Comment (talevy) [inline src/main/java/org/logstash/beats/BeatsHandler.java]: do we have tests for this timeout?

**Comment by ph** [comment]
> I have a question about the keepAlive. what happens if there are no messages to be processed and we close the connection? will filebeat properly handle the networking in this scenario?

They will get a EOF, and it will reconnect maybe we should change the default values for 15 minutes instead of seconds? what do you think?.

File: src/main/java/org/logstash/beats/BeatsHandler.java (hunk 1 above)
Comment (ph) [inline src/main/java/org/logstash/beats/BeatsHandler.java]: We have test to cover that will check that a connection is close when the timeout is reached its in the `ServerTest`

**Comment by talevy** [comment]
> maybe we should change the default values for 15 minutes instead of seconds? what do you think?.

maybe default to 1 minute?

**Comment by ph** [comment]
Will make that change to 1 minute.

**Comment by talevy** [comment]
LGTM

**Pull request closed without merging**